### DISCUSSION TITLE
Error when using Paper from MUI with prerender and only in Production

### DISCUSSION THREAD

**Discussion by teukuridho** [vikejs/vike#1573] [Discussion] [answered]
I got the following error when using Paper from MUI after building for production with setting `prerender: true`
```
(node:12166) Warning: To load an ES module, set "type": "module" in the package.json or use the .mjs extension.
(Use `node --trace-warnings ...` to show where the warning was created)
/Users/teukuridho/Documents/Works/PersonalWorks/vike-unilever-ert/node_modules/.pnpm/@mui+material@5.15.13_@emotion+react@11.11.4_@emotion+styled@11.11.0_@types+react@18.2.66_react-dom@18.2.0_react@18.2.0/node_modules/@mui/material/Paper/index.js:3
export { default } from './Paper';
^^^^^^

SyntaxError: Unexpected token 'export'
```

This is the piece of my code
```
import Paper from '@mui/material/Paper';
...
<TableContainer component={Paper}>
                <Table>
                    <TableHead>
                        <TableRow>
                            <TableCell>No</TableCell>
...
```

I have followed `https://vike.dev/broken-npm-package#workaround`;

Change exports
```
import PaperXD from '@mui/material/Paper';
// @ts-ignore
const Paper = PaperXD.default
```

Settings cjsInterop
```
cjsInterop({
        dependencies: [
          // Add problematic npm package here
          "@mui/icons-material/Menu",
          "@mui/material/Paper",
          "@mui/icons-material/Edit",
        ]
    }),
```

Settings noExternal
```
ssr: {
    noExternal: ['@mui/material/Paper']
  }
```

Nothing works; how do I resolve this? thanks 🙏

**Comment by brillout** [comment]
Minimal reproduction welcome and I'll have a look at it.

**Comment by teukuridho** [comment]
well this is kinda silly, 
In my main project, I imported the Paper with `import Paper from '@mui/material/Paper';` based on MUI documentation and got the error with prerendered production
In test project I imported this way `import { Paper } from "@mui/material";` and it ran without error with prerendered production

don't know how this works, anyway if you want to have a look, this is the repo: [https://github.com/teukuridho/vike-react-paper-mui-test](https://github.com/teukuridho/vike-react-paper-mui-test)

**Reply by brillout** [reply]
I need the reproduction to be minimal.

**Reply by teukuridho** [reply]
1. `import Paper from '@mui/material/Paper'`
2. use it like this ```<Paper>Test</Paper>```
3. set `ssr({
        prerender: true
      }),` and `cjsInterop({
        dependencies: [
          "@mui/material/Paper",
        ]
      }),`  in `vite.config.ts`
4. run `vite build`
5. got the following error 
```
(node:51236) Warning: To load an ES module, set "type": "module" in the package.json or use the .mjs extension.
export { default } from './Paper';
SyntaxError: Unexpected token 'export'
```

**Reply by brillout** [reply]
As a GitHub repo with reproduction steps.

**Reply by teukuridho** [reply]
I have updated the repo [https://github.com/teukuridho/vike-react-paper-mui-test](https://github.com/teukuridho/vike-react-paper-mui-test) with  the reproduction steps

**Reply by brillout** [reply]
https://github.com/brillout/vike-with-mui

**Reply by teukuridho** [reply]
Thank you for taking time.
It works, but when I set `prerender: true` in `ssr` in `vite.config.ts`, it still returns this error
```
(node:58271) Warning: To load an ES module, set "type": "module" in the package.json or use the .mjs extension.
(Use `node --trace-warnings ...` to show where the warning was created)
1.0_@types+react@18.2.69_react-dom@18.2.0_react@18.2.0/node_modules/@mui/material/Paper/index.js:3
export { default } from './Paper';
```

A solution I have found is to properly import it `Paper` like this `import { Paper } from "@mui/material";`, and build and no errors get thrown. Maybe this is just a Vite thing, not Vike

**Reply by brillout** [reply, marked as answer]
I further improved the DX around all that.

Let me know if you still have issues or questions.
